lib/CreepPlayers/Behaviors/Melee.js

```diff
@@ -0,0 +1,30 @@
+
+var MoveUtils = require('../../Utils/MoveUtils');
+var Base = require('./Base');
+
+/**
+ *
+ * @param creepController @type CreepController
+ * @constructor
+ */
+function Melee(creepController) {
+    this.cc = creepController;
+}
+
+Melee.prototype = Object.create(Base.prototype);
+
+Melee.prototype.act = function() {
+    var playerLoc = this.cc.getPlayerInfo().location;
+    var ourLoc = this.cc.getCurrentLocation();
+    var toPlayer = ourLoc.directionTo(playerLoc);
+
+    if (playerLoc.isAdjacentTo(ourLoc)) {
```

Comment:
Not for this behavior, it's just one very simple thing so that it can be re-used for other stuff as well. say monsters that can't move, or ones that want only to melee if the player gets within range, not move there themselves.